python/dgl/transform.py

```diff
@@ -1367,4 +1367,125 @@ def as_immutable_graph(hg):
     g.edata.update(hg.edata)
     return g
 
+def compose(graphs):
+    """Return a composition of the given list of graphs.
+
+    Given the list of graphs with the same set of nodes, ``compose`` will return a new
+    graph with the same set of nodes with all the edges from the list of graphs.
+
+    All edges from all the graphs will be added.
+
+    If the graphs are heterogeneous, the graphs must have the same metagraph.  All edges from
+    all the graphs will be added for each edge type.
+
+    The edges in the first graph will be added first, followed by the edges from the second graph,
+    etc.
+
+    Notes
+    -----
+    All graphs must have the same set of nodes and metagraphs.  An error will be raised otherwise.
+
+    Parameters
+    ----------
+    graphs : list[DGLHeteroGraph]
+        The list of graphs
+
+    Returns
+    -------
+    DGLHeteroGraph
+        The composed graph.
+
+    Examples
+    --------
+    >>> g1 = dgl.graph(([0, 1, 2], [1, 0, 2]), num_nodes=5)
+    >>> g2 = dgl.graph(([1, 3, 3], [0, 2, 3]), num_nodes=5)
+    >>> g = dgl.compose([g1, g2])
+    >>> g.all_edges(order='eid')
+    (tensor([0, 1, 2, 1, 3, 3]), tensor([1, 0, 2, 0, 2, 3]))
+    """
+    # Exposes the UnionCOO/UnionCSR functions to users here
+    pass
+
+def add_reverse(g):
```

Comment:
Other behavior and input arguments I would recommend:
1. `share_ndata=True`. If True, we will perform a shallow copy of the node features (tensor-level, not frame-level) to the new graph.
2. `share_edata=False`. If True, the features for the new edges (j, i) added will be the same as (i, j) in the original graph.

Reply:
If feature inheritance is involved then both `share_ndata` and `share_edata` can be omitted.  I can add the third argument.

Reply:
I'd still prefer making `share_edata` an optional argument as this may not be always desired.

Reply:
By feature inheritance I don't think the feature will be materialized to a tensor.  So why it is not always desired?